internal/sdkprovider/service/project/organization_project.go

```diff
@@ -0,0 +1,276 @@
+	emails, ok := d.GetOk("technical_emails")
+	if ok {
+		var upReq = organizationprojects.OrganizationProjectsUpdateIn{
+			TechEmails: func() *[]organizationprojects.TechEmailIn {
+				var techEmails []organizationprojects.TechEmailIn
+				for _, e := range emails.(*schema.Set).List() {
+					techEmails = append(techEmails, organizationprojects.TechEmailIn{Email: e.(string)})
+				}
+
+				return lo.ToPtr(techEmails)
+			}(),
+		}
+
+		_, err = client.OrganizationProjectsUpdate(ctx, orgID, projectID, &upReq)
+		if err != nil {
+			return err
+		}
+	}
+
+	d.SetId(schemautil.BuildResourceID(orgID, projectID))
+
+	return resourceOrganizationProjectRead(ctx, d, client)
+}
+
+func resourceOrganizationProjectRead(ctx context.Context, d *schema.ResourceData, client avngen.Client) error {
+	var project *organizationprojects.ProjectOut
+
+	orgID, projectID, err := schemautil.SplitResourceID2(d.Id())
+	if err != nil {
+		return fmt.Errorf("error parsing resource ID: %w", err)
+	}
+
+	resp, err := client.OrganizationProjectsList(ctx, orgID)
+	if err != nil {
+		return err
+	}
+
+	for _, p := range resp.Projects {
+		if p.ProjectId == projectID {
+			project = &p
+			break
+		}
+	}
+
+	if project == nil {
+		return fmt.Errorf("project not found: %w", schemautil.ResourceReadHandleNotFound(err, d))
+	}
+
+	if err = d.Set("organization_id", orgID); err != nil {
+		return err
+	}
+	if err = d.Set("project_id", project.ProjectId); err != nil {
+		return err
+	}
+	if err = d.Set("billing_group_id", project.BillingGroupId); err != nil {
+		return err
+	}
+	if err = d.Set("parent_id", project.ParentId); err != nil {
+		return err
+	}
+	if err = d.Set("tag", schemautil.SetTagsTerraformProperties(project.Tags)); err != nil {
+		return err
+	}
+
+	var techEmails = make([]string, 0, len(project.TechEmails))
+	for _, e := range project.TechEmails {
+		techEmails = append(techEmails, e.Email)
+	}
+
+	if err = d.Set("technical_emails", techEmails); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceOrganizationProjectUpdate(ctx context.Context, d *schema.ResourceData, client avngen.Client) error {
+	orgID, projectID, err := schemautil.SplitResourceID2(d.Id())
+	if err != nil {
+		return fmt.Errorf("error parsing resource ID: %w", err)
+	}
+
+	// only handle allowed updates (tags, technical_emails, project_id)
+	var updateReq organizationprojects.OrganizationProjectsUpdateIn
+
+	if d.HasChange("project_id") {
+		updateReq.ProjectId = schemautil.OptionalStringPointer(d, "project_id")
+	}
+	if d.HasChange("tag") {
+		updateReq.Tags = lo.ToPtr(schemautil.GetTagsFromSchema(d))
+	}
+	if d.HasChange("technical_emails") {
+		var techEmails []organizationprojects.TechEmailIn
+		for _, e := range d.Get("technical_emails").(*schema.Set).List() {
+			techEmails = append(techEmails, organizationprojects.TechEmailIn{Email: e.(string)})
+		}
+		updateReq.TechEmails = lo.ToPtr(techEmails)
+	}
```

Comment:
nit: I think `HasChange` won't make any difference here. I believe, putting values directly would do the same job, and would be less confusing.

Reply:
It's here just to prevent sending redundant data in the requests if it is not needed.